PR for SRVCOM-3999: Add a new “Integrating Service Mesh 3.x with OpenShift Serverless" section in the serverless docs (#101668)
openshift/openshift-docs: modules/serverless-ossm-external-certs.adoc

.Prerequisites

* You have cluster administrator permissions on {ocp-product-title}, or you have cluster or dedicated administrator permissions on {rosa-product-title} or {dedicated-product-title}.
* You have installed the {ServerlessOperatorName} and Knative Serving.
* You have installed the OpenShift CLI (`oc`).

> * The `knative-serving-ingress` namespace exists, or the {ServerlessOperatorName} has been installed (which automatically creates this namespace).

Reviewer: ```suggestion
* You have the `knative-serving-ingress` namespace, or you have installed the {ServerlessOperatorName}, which automatically creates this namespace.
```
Changing sentence structure for parallelism.

> * You have the `knative-serving-ingress` namespace, or you have installed the {ServerlessOperatorName}, which automatically creates this namespace.

Reviewer 2: "You have the `knative-serving-ingress` namespace" - This is a little unclear to me, can we rephrase it a little?

    -in wildcard.csr \
    -out wildcard.crt
----
. Create a secret containing the wildcard certificate in the `knative-serving-ingress` namespace:
+
[source,terminal]
----

> $ oc create -n knative-serving-ingress secret tls wildcard-certs \

Reviewer 3: That is the best approach in my opinion.

> $ oc create -n knative-serving-ingress secret tls wildcard-certs \

Author: so should I add my old changes back related to OSSM 2 and create a separate for for OSSM 3? 

As the size of this PR is `XL` now and we can not go beyond XL :|

    -in wildcard.csr \
    -out wildcard.crt
----
. Create a secret containing the wildcard certificate by entering one of the following commands, depending on your {SMProductShortName} version:

** Option A: For {SMProductShortName} 2.x, create the secret in the `istio-system` namespace by entering the folloing command:
+
[source,terminal]
----
$ oc create -n istio-system secret tls wildcard-certs \
    --key=wildcard.key \
    --cert=wildcard.crt
----

** Option B: For {SMProductShortName} 3.x, create the secret in the `knative-serving-ingress` namespace by entering the folloing command:
+
[source,terminal]
----
$ oc create -n knative-serving-ingress secret tls wildcard-certs \
    --key=wildcard.key \
    --cert=wildcard.crt
----
+

> The namespace used for the secret depends on the version of {SMProductShortName}. {SMProductShortName} 2.x expects the certificate in the `istio-system` namespace, while {SMProductShortName} 3.x uses the dedicated `knative-serving-ingress` namespace, where the {ServerlessProductName} ingress gateway runs.

Reviewer: ```suggestion
The namespace used for the secret depends on the version of {SMProductShortName}. {SMProductShortName} 2.x expects the certificate in the `istio-system` namespace. The {SMProductShortName} 3.x uses the dedicated `knative-serving-ingress` namespace where the {ServerlessProductName} ingress gateway runs.
```
